test/unit/lib/challenge.test.ts

```diff
@@ -26,3 +29,226 @@ describe('Challenge Generation', (): void => {
+  })
+
+  // Using a hardcoded key, challenge and sign triplet
+  it('returns false for signature based on wrong challenge - hardcoded EC Key (secp256k1)', (): void => {
+    const { message, keyPair } = Credential.EC
+
+    // Need another Sign object instead of updating the outer one
+    // because of parallel test runs
+    const anotherChallenge = 'This is a different message'
+    const anotherSigner = crypto.createSign('SHA256')
+
+    anotherSigner.update(anotherChallenge)
+
+    const sign = signer.sign(keyPair.private, 'base64')
+    const verified = verifySign(message, sign, keyPair.public)
+
+    expect(verified).toEqual(false)
+  })
+
+  it('verifies correct signature - RSA 2048 Key', (): void => {
+    const realKeyPair = crypto.generateKeyPairSync('rsa', {
+      modulusLength: 2048 // Key length in bits
+    })
+
+    const sign = signer.sign(realKeyPair.privateKey, 'base64')
+    const verified = verifySign(challenge, sign, realKeyPair.publicKey)
+
+    expect(verified).toEqual(true)
+  })
+
+  // Using a correct hardcoded key, challenge and sign triplet
+  it('verifies correct signature - hardcoded RSA 2048 key', (): void => {
+    const { message, signature, keyPair } = Credential.RSA
+
+    const verified = verifySign(message, signature, keyPair.public)
+
+    expect(verified).toEqual(true)
+  })
+
+  it('returns false on signature based on wrong key - RSA 2048 Key', (): void => {
+    const fakeKeyPair = crypto.generateKeyPairSync('rsa', {
+      modulusLength: 2048 // Key length in bits
+    })
+
+    const realKeyPair = crypto.generateKeyPairSync('rsa', {
+      modulusLength: 2048 // Key length in bits
+    })
+
+    const sign = signer.sign(fakeKeyPair.privateKey, 'base64')
+    const verified = verifySign(challenge, sign, realKeyPair.publicKey)
+
+    expect(verified).toEqual(false)
+  })
+
+  // Using a hardcoded key, challenge and sign triplet
+  it('returns false for signature based on wrong challenge - hardcoded RSA 2048 key', (): void => {
+    const { message, keyPair } = Credential.RSA
+
+    // Need another Sign object instead of updating the outer one
+    // because of parallel test runs
+    const anotherChallenge = 'This is a different message'
+    const anotherSigner = crypto.createSign('SHA256')
+
+    anotherSigner.update(anotherChallenge)
+
+    const sign = signer.sign(keyPair.private, 'base64')
+    const verified = verifySign(message, sign, keyPair.public)
+
+    expect(verified).toEqual(false)
+  })
+
+  it('properly uses crypto functions and handles exceptions', (): void => {
+    const { message, keyPair } = Credential.RSA
+
+    // Setting up mocks
+    const createVerifySpy = jest.spyOn(crypto, 'createVerify').mockImplementationOnce((): crypto.Verify => {
+      throw new Error('Unable to create Verify in mock')
+    })
+
+    const loggerPushSpy = jest.spyOn(Logger, 'push')
+    const loggerErrorSpy = jest.spyOn(Logger, 'error').mockImplementation((): void => {})
```

Comment:
```suggestion
    // we mock `Logger.error` because it indirectly calls Logger.push and we want to only test usage of Logger's interface
    const loggerErrorSpy = jest.spyOn(Logger, 'error').mockImplementation((): void => {})
```